recipes/zbar/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+
+class zbarConan(ConanFile):
+    name = "zbar"
+    license = "LGPL-2.1-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://zbar.sourceforge.net/"
+    topics = ("conan", "zbar", "bar codes")
+    description = "ZBar is an open source software suite for reading bar codes\
+                   from various sources, such as video streams, image files and raw intensity sensors"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "without_video": [True, False],
+               "without_imagemagick": [True, False],
+               "without_gtk": [True, False],
+               "without_qt": [True, False],
+               "without_python_bindings": [True, False],
+               "with_x": [True, False],
+               "without_xshm": [True, False],
+               "without_xv": [True, False],
+               "without_jpeg": [True, False],
+               "disable_pthread": [True, False],}
+    default_options = {'shared': False,
+                       'fPIC': True,
+                       'without_video': True,
+                       'without_imagemagick': True,
+                       'without_gtk': True,
+                       'without_qt': True,
+                       'without_python_bindings': True,
+                       'with_x': False,
+                       'without_xshm': False,
+                       'without_xv': False,
+                       'without_jpeg': False,
+                       'disable_pthread': False,
+                       }
+
+    _env_build = None
+    _env_args = []
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_autotools(self):
+        if not self._env_build:
+            self._env_build = AutoToolsBuildEnvironment(self)
+            if self.options.without_video:
+                self._env_args.extend(["--disable-video"])
+            if self.options.without_imagemagick:
```

Comment:
these options with/without, I think you should force both cases, not only when conan option is true

Reply:
What do you mean by force both cases?

Reply:
```python
    self._env_args.extend(["--without-imagemagick" if self.options.without_imagemagick else "--with-imagemagick"])
```
(if the configure script accepts both type of arguments)

Reply:
Got it.

Reply:
If you do this for all arguments, you can simplify the arguments as:
```python
self._env_args.extend([
    "--with-python" if self.options.with_python else "--without-python",
    "--with-x" if self.options.with_x else "--without-x",
    "--with-xshm" if self.options.with_x else "--without-xshm",
    "--with-xv" if self.options.with_xv else "--without-xv",
    "--with-enable-pthread" if self.options.enable_pthread else "--disable-pthread",
])
```
Imho, cleaner code.